src/libAtomVM/jit.c

```diff
@@ -83,6 +83,21 @@ _Static_assert(offsetof(Context, bs_offset) == 0xD0, "ctx->bs_offset is 0xD0 in
 _Static_assert(offsetof(JITState, module) == 0x0, "jit_state->module is 0x0 in jit/src/jit_aarch64.erl");
 _Static_assert(offsetof(JITState, continuation) == 0x8, "jit_state->continuation is 0x8 in jit/src/jit_aarch64.erl");
 _Static_assert(offsetof(JITState, remaining_reductions) == 0x10, "jit_state->remaining_reductions is 0x10 in jit/src/jit_aarch64.erl");
+#elif JIT_ARCH_TARGET == JIT_ARCH_ARMV6M
```

Comment:
I suggest a strategy to reduce duplication here: most of the times offsets are always the same, the only thing really matters is whether arch is 32-bit or 64-bit.
Some odd architectures might have some specific paddings, but I don't think will happen soon.
So we might do:
```c
#if JIT_ARCH_TARGET == JIT_ARCH_WITH_ODD_PADDING
#elif JIT_ARCH_TARGET_BITS == 64
#elif JIT_ARCH_TARGET_BITS == 32
#endif
```

For this specific refactor feel free to add a commit on top of this branch, or to make a new PR, or anything that works for you.